homeassistant/components/evohome/__init__.py

```diff
@@ -44,229 +44,347 @@
-            "Check your network and the vendor's status page. "
-            "Resolve any errors and restart HA."
+        # this appears to be common with Honeywell's servers
+        _LOGGER.warning(
+            "Unable to connect with the vendor's server. "
+            "Check your network and the vendor's status page."
+            "Message is: %s",
+            err
         )
         return False
 
-    finally:  # Redact any config data that's no longer needed
-        for parameter in CONF_SECRETS:
-            evo_data['params'][parameter] = 'REDACTED' \
-                if evo_data['params'][parameter] else None
+    except requests.exceptions.HTTPError:
+        if err.response.status_code == HTTP_SERVICE_UNAVAILABLE:
+            _LOGGER.warning(
+                "Vendor says their server is currently unavailable. "
+                "Check the vendor's status page."
+            )
+            return False
 
-    evo_data['status'] = {}
+        if err.response.status_code == HTTP_TOO_MANY_REQUESTS:
+            _LOGGER.warning(
+                "The vendor's API rate limit has been exceeded. "
+                "Consider increasing the %s.", CONF_SCAN_INTERVAL
+            )
+            return False
 
-    # Redact any installation data that's no longer needed
-    for loc in client.installation_info:
-        loc['locationInfo']['locationId'] = 'REDACTED'
-        loc['locationInfo']['locationOwner'] = 'REDACTED'
-        loc['locationInfo']['streetAddress'] = 'REDACTED'
-        loc['locationInfo']['city'] = 'REDACTED'
-        loc[GWS][0]['gatewayInfo'] = 'REDACTED'
+        raise  # we don't expect/handle any other HTTPErrors
 
-    # Pull down the installation configuration
-    loc_idx = evo_data['params'][CONF_LOCATION_IDX]
-    try:
-        evo_data['config'] = client.installation_info[loc_idx]
 
-    except IndexError:
-        _LOGGER.error(
-            "setup(): config error, '%s' = %s, but its valid range is 0-%s. "
-            "Unable to continue. Fix any configuration errors and restart HA.",
-            CONF_LOCATION_IDX, loc_idx, len(client.installation_info) - 1
-        )
+async def async_setup(hass, hass_config):
+    """Create a (EMEA/EU-based) Honeywell evohome system."""
+    broker = EvoBroker(hass, hass_config[DOMAIN])
+    if not await broker.init_client():
         return False
 
-    if _LOGGER.isEnabledFor(logging.DEBUG):
-        tmp_loc = dict(evo_data['config'])
-        tmp_loc['locationInfo']['postcode'] = 'REDACTED'
+    load_platform(hass, 'climate', DOMAIN, {}, hass_config)
+    if broker.tcs.hotwater:
+        load_platform(hass, 'water_heater', DOMAIN, {}, hass_config)
 
-        if 'dhw' in tmp_loc[GWS][0][TCS][0]:  # if this location has DHW...
-            tmp_loc[GWS][0][TCS][0]['dhw'] = '...'
+    async_track_time_interval(
+        hass,
+        broker.update,
+#       hass_config[DOMAIN][CONF_SCAN_INTERVAL]
+        timedelta(seconds=10)  # TODO: delete me
```

Comment:
TODO

Reply:
Yeah, sorry - I am still in dev mode hence the **WIP:** in the title.

I though if I had **WIP:** in the title, then people would know the PR wasn't ready for review? But I had one merged!!